frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx

```diff
@@ -120,6 +121,19 @@ function MainNavbarView({
     ({ id, can_write }) => (id === null || id === "root") && can_write,
   );
 
+  const [collectionsWithoutTrash, trashCollection] = useMemo(() => {
+    let trashCollection;
+    const collectionsWithoutTrash = [];
+    for (const c of collections) {
+      if (c.type === "trash") {
+        trashCollection = c;
+      } else {
+        collectionsWithoutTrash.push(c);
+      }
+    }
+    return [collectionsWithoutTrash, trashCollection];
```

Comment:
```suggestion
    return _.partition(collections, (c) => c.type !== "trash");
```

and the trash will already be in an array so you can pass it straight to the tree component below